homeassistant/components/ads/cover.py

```diff
@@ -0,0 +1,175 @@
+                 ads_var_pos_set, ads_var_open,
+                 ads_var_close, ads_var_stop, name, device_class):
+        """Initialize AdsCover entity."""
+        super().__init__(ads_hub, name, ads_var_is_closed)
+        if self._ads_var is None:
+            if ads_var_position is not None:
+                self._unique_id = ads_var_position
+            elif ads_var_pos_set is not None:
+                self._unique_id = ads_var_pos_set
+            elif ads_var_open is not None:
+                self._unique_id = ads_var_open
+
+        self._state_dict[STATE_KEY_POSITION] = None
+        self._ads_var_position = ads_var_position
+        self._ads_var_pos_set = ads_var_pos_set
+        self._ads_var_open = ads_var_open
+        self._ads_var_close = ads_var_close
+        self._ads_var_stop = ads_var_stop
+        self._device_class = device_class
+
+    async def async_added_to_hass(self):
+        """Register device notification."""
+        if self._ads_var is not None:
+            await self.async_initialize_device(self._ads_var,
+                                               self._ads_hub.PLCTYPE_BOOL)
+
+        if self._ads_var_position is not None:
+            await self.async_initialize_device(self._ads_var_position,
+                                               self._ads_hub.PLCTYPE_BYTE,
+                                               STATE_KEY_POSITION)
+
+    @property
+    def state(self):
+        """Return the state of the cover."""
+        closed = self.is_closed
+
+        if closed is None:
+            return None
+
+        return STATE_CLOSED if closed else STATE_OPEN
+
+    @property
+    def device_class(self):
+        """Return the class of this cover."""
+        return self._device_class
+
+    @property
+    def is_closed(self):
+        """Return if the cover is closed."""
+        if self._ads_var is not None:
+            return self._state_dict[STATE_KEY_STATE]
+        if self._ads_var_position is not None:
+            return self._state_dict[STATE_KEY_POSITION] == 0
+        return None
+
+    @property
+    def current_cover_position(self):
+        """Return current position of cover."""
+        return self._state_dict[STATE_KEY_POSITION]
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        supported_features = SUPPORT_OPEN | SUPPORT_CLOSE
+
+        if self._ads_var_stop is not None:
+            supported_features |= SUPPORT_STOP
+
+        if self._ads_var_pos_set is not None:
+            supported_features |= SUPPORT_SET_POSITION
+
+        return supported_features
+
+    def stop_cover(self, **kwargs):
+        """Fire the stop action."""
+        if self._ads_var_stop:
+            self._ads_hub.write_by_name(self._ads_var_stop, True,
+                                        self._ads_hub.PLCTYPE_BOOL)
+
+    def set_cover_position(self, **kwargs):
+        """Set cover position."""
+        position = kwargs.get(ATTR_POSITION)
```

Comment:
Use `dict[key]`. Position is always present as its validated by he service schema.